### PR TITLE
Fix Broken UI Test for all Admin Configurable Settings 

### PR DESCRIPTION
**Pull request by Stryker101**
# **This is a pull request raised for issue https://github.com/odk-x/tool-suite-X/issues/470**

## **Reason for Fix:**
There are configurable settings available for a normal user of the app and there are configurable settings available for a user with admin privileges. The UI tests for these settings screen was broken following a rework of the settings screen. This PR was raised to fix the broken UI Test for the configurable settings that are available to a user with ADMIN privileges. These tests includes the following:

 -AdminConfigurableServerSettingsFragmentTest.java
 -AdminConfigurableDeviceSettingsFragmentTest.java
 -AdminConfigurableTablesSettingsFragmentTest.java

## **What was changed in these files:**

AdminConfigurableServerSettingsFragmentTest.java
--Nothing changed. this test was fixed when the enableAdminMode(); function in BaseUItest was fixed

AdminConfigurableDeviceSettingsFragmentTest.java
--corrected the reference for the string to be matched.

AdminConfigurableTablesSettingsFragmentTest.java
--corrected the reference for the string to be matched.

# ***Also updated the PR with a fix for VerifyUserPermissionTest.java**

## **What was changed in these files:**

VerifyUserPermissionTest.java
--corrected the reference for the string to be matched.

## ***N/B: Please note that following update of this PR with the fix for VerifyUserPermissionTest.java, this concludes the fixing of all the broken UI tests in the Preferences package. except for the QR code scanner test which has no current implementation.**

Screenshot 2024-03-29 213306
Screenshot 2024-03-29 213423
Screenshot 2024-03-29 213601
Screenshot 2024-03-29 213713

### CODE REVIEW

**Comment by Redeem-Grimm-Satoshi** [comment]
Well done! I see some changes made but I haven't tested it yet. Send a screenshot showing how it works at your end ( test passed ) then I can proceed to review from there

**Comment by Stryker101** [comment]
Hi @Redeem-Grimm-Satoshi I've  added the screenshot(s) as requested.

File: services_app/src/androidTest/java/org/opendatakit/BaseUITest.java [hunk 1]
```diff
@@ -190,14 +190,13 @@ public static ViewAction waitFor(long delay) {
...
         onView(withId(androidx.preference.R.id.recycler_view))
-                .perform(RecyclerViewActions.actionOnItem(hasDescendant(withText(R.string.enable_admin_password)),
+                .perform(RecyclerViewActions.actionOnItem(hasDescendant(withText(R.string.user_restrictions)),
                         click()));
         onView(withId(androidx.preference.R.id.recycler_view))
                 .perform(RecyclerViewActions.actionOnItemAtPosition(1,
                         click()));
-        onView(withText(R.string.change_admin_password))
-                .inRoot(isDialog())
-                .check(matches(isDisplayed()));
```
Comment (Redeem-Grimm-Satoshi) [inline services_app/src/androidTest/java/org/opendatakit/BaseUITest.java:200]: Why remove the password dialog verification check?

File: services_app/src/androidTest/java/org/opendatakit/services/preferences/fragments/AdminConfigurableTablesSettingsFragmentTest.java [hunk 2]
```diff
@@ -100,7 +97,7 @@ private void launchTableServerSettingPreferenceScreen() {
         onView(isRoot()).perform(waitFor(1000));
 
         onView(withId(androidx.preference.R.id.recycler_view))
-                .perform(RecyclerViewActions.actionOnItem(hasDescendant(withText(R.string.tool_tables_settings)),
+                .perform(RecyclerViewActions.actionOnItem(hasDescendant(withText(R.string.odkx_tables)),
```
Comment (Redeem-Grimm-Satoshi) [inline services_app/src/androidTest/java/org/opendatakit/services/preferences/fragments/AdminConfigurableTablesSettingsFragmentTest.java:100]: why change the string `tool_tables_settings` ( Table-specific Settings ) to `odkx_tables`( ODK-X Tables ) ? 

Any reason for that?

**Review by Redeem-Grimm-Satoshi — CHANGES_REQUESTED** [review]
More work needed

File: services_app/src/androidTest/java/org/opendatakit/services/preferences/fragments/AdminConfigurableTablesSettingsFragmentTest.java (hunk 2 above)
Comment (Stryker101) [reply services_app/src/androidTest/java/org/opendatakit/services/preferences/fragments/AdminConfigurableTablesSettingsFragmentTest.java:100]: That's the current string resource for the displayed text.

File: services_app/src/androidTest/java/org/opendatakit/BaseUITest.java (hunk 1 above)
Comment (Stryker101) [reply services_app/src/androidTest/java/org/opendatakit/BaseUITest.java:200]: Bacause that view isn't there anymore. Check for what isn't there is what broke that test. We can get on a skype call and run these test together.

**Comment by Stryker101** [comment]
> More work needed

Those questions have been answered.

**Comment by Redeem-Grimm-Satoshi** [comment]
> > More work needed
> 
> Those questions have been answered.

I don't think you've understood me well. Answer the questions I asked up, It's that simple!

**Comment by Stryker101** [comment]
> > > More work needed
> > 
> > Those questions have been answered.
> 
> I don't think you've understood me well. Answer the questions I asked up, It's that simple!

If you think i misunderstood you , i think the productive thing to do is explain what you think i missed. I gave a response based on how i understood your previous comment.

@wbrunette can you weigh in on this please?

**Review by Redeem-Grimm-Satoshi — APPROVED** [review]

**Review by wbrunette — APPROVED** [review]

**Pull request merged**